Jenkinsfile

```diff
@@ -113,7 +113,7 @@ def cmake_build(Map conf=[:]){
 def getDockerImageName(prefixpath)
 {
     def branch =  sh(script: "echo ${scm.branches[0].name} | sed 's/[^a-zA-Z0-9]/_/g' ", returnStdout: true).trim()
-    def image = "${env.MIOPEN_IMAGE_URL}"
+    def image = "${env.MIOPEN_IMAGE_URL}_mlir266ad"
```

Comment:
@atamazov thanks and yes it should be documented. Step (B) can be set manually on Jenkins webpage for the CI docker candidate, if one prefer not to change Jenkinsfile

Reply:
I need a couple of clarifications:
 - (A) This seems to be a manual process. I'd prefer that MIOpen end of CI provides a docker build job dedicated to build image according to a particular branch. This way, we can easily have an URL to refer to that image is successfully built.
 - (B) This step seems unnecessary because ultimately changing the docker environment requires a pull request. A dependency such as rocMLIR doesn't have to have two passing record: one for develop branch and another for PR. (Occasionally when MIOpen change is required, the new docker environment would not work with the pre-existing develop branch)
 - (C) Sounds good, so we basically manually trigger CI and fill in that `DOCKER_IMAGE_OVERRIDE`. However, the current implementation of this seems to be always reverting to an empty string after each run, which seems to be a big inconvenience and can potentially waste MIOpen CI resources. A developer need to manually go and kick off CI with the override for every commit.

> When all tests pass, MIOPEN_IMAGE_URL environment variable can be globally updated for the whole CI.

Where do we set this? I only see this variable referenced from dockerfile but didn't see it elsewhere in the code base.

Reply:
@jerryyin Yes, my proposal assumes that some manual work is necessary when a developer wants to update CI docker image. However, it seems workable out of the box, and we can begin using it right away. The process can be improved/better automated in the future. Let's figure out the drawbacks clearly and open a ticket.

> (A) This seems to be a manual process. I'd prefer that MIOpen end of CI provides a docker build job dedicated to build image according to a particular branch. This way, we can easily have an URL to refer to that image is successfully built.

Building docker image on each CI job could be a waste of resources, because the majority of PRs do not affect the image. But we can add the "Docker Build" step to the Jenkinsfile (and make it switched off by default). A developer can enable this step manually, then download the built image (artifact) from CI, tag it properly and upload to dockerhub.
 
> (B) This step seems unnecessary because ultimately changing the docker environment requires a pull request...

AFAIU pull request is not required to change the docker environment. It is enough to upload a new docker image to `rocm/miopen:ci`, or upload some other docker image and change `MIOPEN_IMAGE_URL` env variable on the nodes.

The purpose of (B) is to check if the new docker image can be used on CI with current develop. This could be necessary, for example, if we decide to revert the PR due to some reason.

> (C) Sounds good, so we basically manually trigger CI and fill in that DOCKER_IMAGE_OVERRIDE. However... a developer need to manually go and kick off CI with the override for every commit.

When (B) and (C) pass, a developer should update `MIOPEN_IMAGE_URL` and then all the further jobs will use the new image. A developer need to manually trigger CI only twice, during steps (B) and (C).

>> MIOPEN_IMAGE_URL environment variable can be globally updated for the whole CI.
> Where do we set this?

IIRC the Jenkins' Web UI should allow this, but I can't give a definitive answer because I do not have access to our CI as of today. I believe that @JehandadKhan or @junliume know the exact answer.

Reply:
@atamazov Agreed, we'll use the manual way to bump docker environment in this PR, and improve it over time.

> we can add the "Docker Build" step to the Jenkinsfile (and make it switched off by default). A developer can enable this step manually

Sounds prefect. Yes, I was thinking about another jenkins job that tracks dependency changes but I think your proposal is better. This is just so that we have the passing record of the docker build somewhere and can refer to when things go wrong.

> AFAIU pull request is not required to change the docker environment.

Just speaking from the rocMLIR integration, we have a commit tracking from `requirements.txt` and it will be very bad if the docker used a different commit id with the one from the `requirements.txt`. So MIOpen should not tolerate us implicitly "swap out" a docker environment that used a different commit#. I think I get it now the purpose of step (B) is to make it clear about whether a dependency can have breaking changes. Since rocMLIR is backward compatible with MIOpen this seems a reasonable step to have.

> a developer should update MIOPEN_IMAGE_URL and then all the further jobs will use the new image

I actually cannot find this `MIOPEN_IMAGE_URL` field but instead can only find `DOCKER_IMAGE_OVERRIDE`. So it'd be a nice clarification if there's a way to persist this to an environment variable.